### PR TITLE
Adding remove query function

### PR DESCRIPTION
**Pull request by RedStylzZ**

### CODE REVIEW

File: backend/src/main/java/com/github/redstylzz/backend/model/Payment.java [hunk 1]
```diff
@@ -54,7 +77,7 @@ public static PaymentDTO convertPaymentToDTO(Payment payment) {
                 .categoryID(payment.categoryID)
                 .description(payment.description)
                 .amount(payment.amount)
-                .payDate(payment.payDate)
+                .payDate(LocalDateTime.from(payment.payDate.atZone(ZoneId.of("GMT+1"))))
```
Comment (synt51) [inline backend/src/main/java/com/github/redstylzz/backend/model/Payment.java:80]: Why are you working with 'LocalDateTime' AND 'Instant'?

File: frontend/src/components/SeriesItem.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    return (
+        <div className={"seriesItemCard"}>
+            <input type="button" onClick={() => deleteSeries(series.seriesId)} value={"Delete"}/>
+            <div className={"seriesItem"}>
+                <h2>Start</h2>
+                <h3>{series.startDate ? <FormatDate date={series.startDate}/> : "Infinite"}</h3>
+                <h2>End</h2>
+                <h3>{series.endDate ? <FormatDate date={series.endDate}/> : "Infinite"}</h3>
+                <h2>Scheduled Day</h2>
+                <h3>{series.scheduledDate}</h3>
```
Comment (synt51) [inline frontend/src/components/SeriesItem.tsx:21]: Coach Jan commented a similar part in my last PR that this type of heading usage breaks the heading semantics.

**Review by synt51 — COMMENTED** [review]
LGTM

**Review by synt51 — APPROVED** [review]
LGTM

File: backend/src/main/java/com/github/redstylzz/backend/controller/PaymentSeriesController.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    private final PaymentSeriesService service;
+
+    private MongoUser getUser(Principal principal) throws ResponseStatusException {
+        try {
+            return userService.getUserByPrincipal(principal);
+        } catch (UsernameNotFoundException e) {
+            LOG.warn("No user found");
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "No user found");
+        }
+    }
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/github/redstylzz/backend/controller/PaymentSeriesController.java:34]: Is this realy necessary? 🤔 We did literally the same thing in the JwtAuthFilter right?

File: backend/src/main/java/com/github/redstylzz/backend/service/ScheduleService.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        LOG.info("Adding payments from series");
+        List<PaymentSeries> series = repository.getAllByScheduledDate(LocalDateTime.now().getDayOfMonth());
+        if (!series.isEmpty()) {
+            List<Payment> payments = series
+                    .stream()
+                    .map(s -> Payment.convertDTOtoPayment(s.getPayment(), s.getUserId(), LocalDate.now().atStartOfDay().toInstant(ZoneOffset.UTC)))
+                    .toList();
+            paymentRepository.saveAll(payments);
+        }
+    }
```
Comment (Flooooooooooorian) [inline backend/src/main/java/com/github/redstylzz/backend/service/ScheduleService.java:39]: Interesting way of solving this. I would think about not creating any Payments for the PaymentSeries and instead when user wants all his payments give him the payments and the series, and calculate based on the series which payment already happen in the series...? 🤔

File: frontend/src/pages/SchedulePage.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    useEffect(() => {
+        seriesController.getSeries().then(setSeries)
+        categoryController.getCategories().then(setCategories)
+    }, [seriesController, categoryController])
+
+    const addSeries = (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+        const form = event.currentTarget
+        const formElements = form.elements as typeof form.elements & SelectInput
+        const categoryID: string = formElements.selectCategory.value
```
Comment (Flooooooooooorian) [inline frontend/src/pages/SchedulePage.tsx:40]: I would save the selected category in a state when the user selects one then you can get it from there and dont need to extract it from the fromEvent

**Review by Flooooooooooorian — APPROVED** [review]

File: frontend/src/pages/SchedulePage.tsx (hunk 5 above)
Comment (RedStylzZ) [reply frontend/src/pages/SchedulePage.tsx:40]: I did that in the first step, but it made problems on the default value

**Pull request merged**